pkg/tests/dbaas_operator_tests.go

```diff
@@ -55,20 +55,9 @@ var _ = Describe("Rhoda e2e Test", func() {
 			for i := range providers {
 				provider := providers[i]
 				providerType := string(provider.SecretData["providerType"])
-				BeforeEach(func() {
-					By(fmt.Sprintf("Checking if provider operator is ready: %s", provider.ProviderName))
-					Eventually(func() bool {
-						if err := client.Get(context.Background(), k8sClient.ObjectKey{
-							Name: providerType,
-						}, &dbaasv1alpha1.DBaaSProvider{}); err != nil {
-							return false
-						}
-						return true
-					}, time.Second*120, time.Second).Should(BeTrue())
-				})
 				It("Should pass when secret and provider created and connection status is checked for "+provider.ProviderName, func() {
 					DeferCleanup(func() {
-						fmt.Println("DeferCleanup Started")
+						By("Deleting Secret: " + provider.SecretName)
 						fmt.Println("deleting Secret: " + provider.SecretName)
```

Comment:
Is this now redundant?

Reply:
Yes, 'By' does print the info, it also provides more detailed logs if it will fails. From ginko site: "If a test succeeds you won't see any output beyond Ginkgo's green dot. If a test fails, however, you will see each step printed out up to the step immediately preceding the failure. "

Reply:
Please run the test so that it everything passes and copy/paste the log output here

Reply:
=== RUN   TestDbaaSOperatorTestHarness
Running getConfig
crunchy
mongo
crdb
Running Suite: DBaaS Operator Test Harness - /Users/olgalavtar/repos/dbaas-e2e-test-harness
===========================================================================================
Random Seed: 1657323482

Will run 5 of 5 specs
checking operator installation
Running getConfig
•Creating secret for : crunchy
Creating inventory for : crunchy
Checking status for : crunchy
Checking status for : crunchy
provider-acct-test-e2e-crunchy
checking DBaaSConnection status for: crunchy-test-instance-olga
checking DBaaSConnection status for: crunchy-test-instance-olga
deleting Secret: dbaas-secret-e2e-crunchy
deleting Connection and Provider for: crunchy
crunchy-test-instance-olga
deleting dbaas connection: crunchy-test-instance-olga
deleting provider Acct: provider-acct-test-e2e-crunchy

------------------------------
• [SLOW TEST] [13.610 seconds]
Rhoda e2e Test Test all the providers Should pass when secret and provider created and connection status is checked for crunchy
/Users/olgalavtar/repos/dbaas-e2e-test-harness/pkg/tests/dbaas_operator_tests.go:79
------------------------------
Creating secret for : mongo
Creating inventory for : mongo
Checking status for : mongo
Checking status for : mongo
provider-acct-test-e2e-mongo
No instances to connect
deleting Secret: dbaas-secret-e2e-mongo
deleting Connection and Provider for: mongo
deleting provider Acct: provider-acct-test-e2e-mongo
• [SLOW TEST] [7.412 seconds]
Rhoda e2e Test Test all the providers Should pass when secret and provider created and connection status is checked for mongo
/Users/olgalavtar/repos/dbaas-e2e-test-harness/pkg/tests/dbaas_operator_tests.go:79
------------------------------
Creating secret for : crdb
Creating inventory for : crdb
Checking status for : crdb
Checking status for : crdb
provider-acct-test-e2e-crdb
checking DBaaSConnection status for: dev-inst-0526-001
checking DBaaSConnection status for: dev-inst-0526-001
deleting Secret: dbaas-secret-e2e-crdb
deleting Connection and Provider for: crdb
dev-inst-0526-001
deleting dbaas connection: dev-inst-0526-001
deleting provider Acct: provider-acct-test-e2e-crdb
• [SLOW TEST] [13.721 seconds]
Rhoda e2e Test Test all the providers Should pass when secret and provider created and connection status is checked for crdb
/Users/olgalavtar/repos/dbaas-e2e-test-harness/pkg/tests/dbaas_operator_tests.go:79
------------------------------
Login
Running getConfig
console-openshift-console.apps.rhoda-sp-prod.lue0.p1.openshiftapps.com
Found Data Services button
Database Access
/k8s/ns/redhat-dbaas-operator/rhoda-admin-dashboard
https://console-openshift-console.apps.rhoda-sp-prod.lue0.p1.openshiftapps.com/k8s/ns/redhat-dbaas-operator/rhoda-admin-dashboard
Database Access 
• [SLOW TEST] [29.680 seconds]
Rhoda e2e Test Console Login UI and pages access check
/Users/olgalavtar/repos/dbaas-e2e-test-harness/pkg/tests/dbaas_operator_tests.go:244
------------------------------
[ReportAfterSuite] [FAILED] [0.000 seconds]
[ReportAfterSuite] Autogenerated ReportAfterSuite for --junit-report
autogenerated by Ginkgo

  Failed to generate JUnit report:
  open /test-run-results/junit-dbaas-operator.xml: no such file or directory
  In [ReportAfterSuite] at: /Users/olgalavtar/go/pkg/mod/github.com/onsi/ginkgo/v2@v2.1.4/reporting_dsl.go:127

  Full Stack Trace
------------------------------

Ran 5 of 5 Specs in 67.153 seconds
FAIL! -- 5 Passed | 0 Failed | 0 Pending | 0 Skipped
    dbaas_e2e_test_harness_suite_test.go:31: error while writing metadata: open /test-run-results/addon-metadata.json: no such file or directory
--- FAIL: TestDbaaSOperatorTestHarness (71.38s)

FAIL

Process finished with the exit code 1

Reply:
RUN   TestDbaaSOperatorTestHarness
Running getConfig
crunchy
mongo
crdb
Running Suite: DBaaS Operator Test Harness - /Users/olgalavtar/repos/dbaas-e2e-test-harness
===========================================================================================
Random Seed: 1657323482

Will run 5 of 5 specs
checking operator installation
Running getConfig
•Creating secret for : crunchy
Creating inventory for : crunchy
Checking status for : crunchy
Checking status for : crunchy
provider-acct-test-e2e-crunchy
checking DBaaSConnection status for: crunchy-test-instance-olga
checking DBaaSConnection status for: crunchy-test-instance-olga
deleting Secret: dbaas-secret-e2e-crunchy
deleting Connection and Provider for: crunchy
crunchy-test-instance-olga
deleting dbaas connection: crunchy-test-instance-olga
deleting provider Acct: provider-acct-test-e2e-crunchy

------------------------------
• [SLOW TEST] [13.610 seconds]
Rhoda e2e Test Test all the providers Should pass when secret and provider created and connection status is checked for crunchy
/Users/olgalavtar/repos/dbaas-e2e-test-harness/pkg/tests/dbaas_operator_tests.go:79
------------------------------
Creating secret for : mongo
Creating inventory for : mongo
Checking status for : mongo
Checking status for : mongo
provider-acct-test-e2e-mongo
No instances to connect
deleting Secret: dbaas-secret-e2e-mongo
deleting Connection and Provider for: mongo
deleting provider Acct: provider-acct-test-e2e-mongo
• [SLOW TEST] [7.412 seconds]
Rhoda e2e Test Test all the providers Should pass when secret and provider created and connection status is checked for mongo
/Users/olgalavtar/repos/dbaas-e2e-test-harness/pkg/tests/dbaas_operator_tests.go:79
------------------------------
Creating secret for : crdb
Creating inventory for : crdb
Checking status for : crdb
Checking status for : crdb
provider-acct-test-e2e-crdb
checking DBaaSConnection status for: dev-inst-0526-001
checking DBaaSConnection status for: dev-inst-0526-001
deleting Secret: dbaas-secret-e2e-crdb
deleting Connection and Provider for: crdb
dev-inst-0526-001
deleting dbaas connection: dev-inst-0526-001
deleting provider Acct: provider-acct-test-e2e-crdb
• [SLOW TEST] [13.721 seconds]
Rhoda e2e Test Test all the providers Should pass when secret and provider created and connection status is checked for crdb
/Users/olgalavtar/repos/dbaas-e2e-test-harness/pkg/tests/dbaas_operator_tests.go:79
------------------------------
Login
Running getConfig
console-openshift-console.apps.rhoda-sp-prod.lue0.p1.openshiftapps.com
Found Data Services button
Database Access
/k8s/ns/redhat-dbaas-operator/rhoda-admin-dashboard
https://console-openshift-console.apps.rhoda-sp-prod.lue0.p1.openshiftapps.com/k8s/ns/redhat-dbaas-operator/rhoda-admin-dashboard
Database Access 
• [SLOW TEST] [29.680 seconds]
Rhoda e2e Test Console Login UI and pages access check
/Users/olgalavtar/repos/dbaas-e2e-test-harness/pkg/tests/dbaas_operator_tests.go:244
------------------------------
[ReportAfterSuite] [FAILED] [0.000 seconds]
[ReportAfterSuite] Autogenerated ReportAfterSuite for --junit-report
autogenerated by Ginkgo

  Failed to generate JUnit report:
  open /test-run-results/junit-dbaas-operator.xml: no such file or directory
  In [ReportAfterSuite] at: /Users/olgalavtar/go/pkg/mod/github.com/onsi/ginkgo/v2@v2.1.4/reporting_dsl.go:127

  Full Stack Trace
------------------------------

Ran 5 of 5 Specs in 67.153 seconds
FAIL! -- 5 Passed | 0 Failed | 0 Pending | 0 Skipped
    dbaas_e2e_test_harness_suite_test.go:31: error while writing metadata: open /test-run-results/addon-metadata.json: no such file or directory
--- FAIL: TestDbaaSOperatorTestHarness (71.38s)

FAIL

Process finished with the exit code 1

Reply:
Thanks, it looks like we get a meaningful name for every Spec that passes, and then the By() for everyone that fails as you said? In that case, I think we could do with much less noise, especially where there is a By() for that printout already

Reply:
I cleaned up the code and left only major printouts.